Bluetooth Ext Adv:Sync: While simultaneous advertiser are working, and skip is non-zero, sync terminates repeatedly

**Describe the bug**
Using the https://github.com/zephyrproject-rtos/zephyr/pull/41091 (force-pushed on 2 Feb), I tested 10 simultaneous advertiser (AoA tag) with one scanner (AoA locator). The source codes are based on "connection-less direction finding" examples, by adding simultaneous tag support to scanner code. The advertiser are working with extended adv interval = 80ms, legacy adv interval = 150ms. All devices are nRF52833DK board.

**Successful Scenario:** If at receiver side, **"skip" parameter is a non-zero value**, then "synchronization" is made successfully and the probability of losing the sync is low.
```
struct bt_le_per_adv_sync_param param;
param.skip = 1;
param.timeout = 2000;
...
bt_le_per_adv_sync_create(&param, synch);
```
The output log is something like this:
```
try to sync tag1
tag1 synced
try to sync tag2
tag2 synced
try to sync tag3
tag3 synced
try to sync tag4
tag4 synced
try to sync tag5
tag5 synced
try to sync tag6
tag6 synced
try to sync tag7
tag7 synced
try to sync tag8
tag8 synced
try to sync tag9
tag9 synced
try to sync tag10
tag10 synced

(after 30minutes no sync lost was seen)
```

**Failure Scenario:** But if, **"skip" parameter is set to zero**, at the beginning of the program, syncs are made normally like above condition, but **after a while it start to lose the sync** with tags, and it usually must **retries more than once to make the sync again**. in most retries, it terminate eventually exactly after calling the ``bt_le_per_adv_sync_create`` function.
```
struct bt_le_per_adv_sync_param param;
param.skip = 0;
param.timeout = 2000;
...
bt_le_per_adv_sync_create(&param, synch);
```
The output log could be:
```
try to sync tag1
tag1 synced
try to sync tag2
tag2 synced
try to sync tag3
tag3 synced
try to sync tag4
tag4 synced
try to sync tag5
tag5 synced
try to sync tag6
tag6 synced
try to sync tag7
tag7 synced
try to sync tag8
tag8 synced
try to sync tag9
tag9 synced
try to sync tag10
tag10 synced

tag3 sync lost
try to sync tag3

tag3 sync lost
try to sync tag3

tag3 sync lost
try to sync tag3

tag3 sync lost
try to sync tag3
tag3 synced

(the above retries occur for other tags too and it is very common to lose the sync, 
which makes the whole system functionality unstable) 
```

In both cases timeout value is same. I supposed that "skip" parameter can not affect on sync termination as I asked this question in nordic devzone, but seems it is not.
https://devzone.nordicsemi.com/f/nordic-q-a/84232/more-information-about-skip-and-timeout-parameters-of-ble-5-advertising

**To Reproduce**
Steps to reproduce the behavior:
1. use "connection-less direction finding" examples
2. Add simultaneous tag support to the scanner source code 
2. west build -b nrf52833dk_nrf52833
3. west flash
4. See error

**Expected behavior**
The termination is only dependent on "timoute" parameter, changing "skip" should not change the behavior of losing sync between scanner and advertiser.

**Impact**
System is not stable when skip=0 and multiple advertiser try to sync with the scanner.

**Logs and console output**
as you can see in the bug description (above)

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: https://github.com/zephyrproject-rtos/zephyr/pull/41091/commits/9749514d83887d1dedde9966a0dc2a5d401aca97

**Additional context**
In addition to the above problem, sometime this assertion failure happens too
```
ASSERTION FAIL [next] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:649
```

Thanks for your time!

I tried my sample code several times and these "assertion failures" were seen:

Test Condition:
Receiver
Source Code: [direction_finding_connectionless_rx_multiple](https://github.com/saleh-unikie/direction_finding_connectionless_rx_multiple)
Zephyr: [64c2633c90](https://github.com/zephyrproject-rtos/zephyr/pull/42678) on top of 5094a6e08c

Transmitter:
Source code: [direction_finding_connectionless_tx](https://github.com/saleh-unikie/direction_finding_connectionless_tx)
Number of transmitters: 10

Failures:
```
C       Creating Sync:[DEVICE]: 27:83:C5:5E:AF:FB (random) ...success.                                                                 
S       PER_ADV_SYNC[7]: [DEVICE]: 27:83:C5:5E:AF:FB (random) synced, Interval 0x0030 (60 ms), PHY LE 2M                               
ASSERTION FAIL [(ret == 0) || (ret == 2)] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:728
```
Full log: [10_cte_enabled_42678.txt](https://github.com/zephyrproject-rtos/zephyr/files/8049789/10_cte_enabled_42678.txt)

After the above log, I added a print line before when it tries to make enable reception of CTE.

```
XC      Creating sync timed out                                                                                                        
Deleting Periodic Advertising Sync...success                                                                                           
ASSERTION FAIL [e] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:938
```
Full log: [12_cte_enabled_42678.txt](https://github.com/zephyrproject-rtos/zephyr/files/8049797/12_cte_enabled_42678.txt)

```
............T   PER_ADV_SYNC[7]: [DEVICE]: 0D:11:A7:7B:D8:D3 (random) sync terminated                                                  
........                                                                                                                               
ASSERTION FAIL [e] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c:938 
```
Full log: [13_cte_enabled_42678.txt](https://github.com/zephyrproject-rtos/zephyr/files/8049802/13_cte_enabled_42678.txt)

```
S       PER_ADV_SYNC[6]: [DEVICE]: 30:6E:A9:78:50:C1 (random) synced, Interval 0x0030 (60 ms), PHY LE 2M                               
E       Enabling reception of CTE                                                                                                      
...ASSEASSERTION FAIL [err >= 0] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:578
```
Full log: [14_cte_enabled_42678.txt](https://github.com/zephyrproject-rtos/zephyr/files/8049807/14_cte_enabled_42678.txt)

```
S       PER_ADV_SYNC[9]: [DEVICE]: 19:D6:38:93:F9:7F (random) synced, Interval 0x0030 (60 ms), PHY LE 2M                               
E       Enabling reception of CTE                                                                                                      
ASSERTION FAIL [next] @ WEST_TOPDIR/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c:653
```
Full log: [15_cte_enabled_42678.txt](https://github.com/zephyrproject-rtos/zephyr/files/8049846/15_cte_enabled_42678.txt)

--- a/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df.c
+++ b/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df.c
@@ -14,6 +14,7 @@
 #include "util/dbuf.h"
 
 #include "hal/cpu.h"
+#include "hal/ccm.h"
 #include "hal/radio_df.h"
 
 #include "pdu.h"

--- a/subsys/bluetooth/controller/ll_sw/ull_df.c
+++ b/subsys/bluetooth/controller/ll_sw/ull_df.c
@@ -466,7 +466,7 @@ uint8_t ll_df_set_cl_iq_sampling_enable(uint16_t handle,
 		 * (-ENOENT) gracefully.
 		 * Periodic sync lost event also disables the CTE sampling.
 		 */
-		err = ull_sync_slot_update(sync, 0, CTE_LEN_MAX_US);
+		err = ull_sync_slot_update(sync, slot_plus_us, slot_minus_us);
 		LL_ASSERT(err == 0 || err == -ENOENT);
 	}
 

--- a/subsys/bluetooth/controller/ll_sw/ull_sync.c
+++ b/subsys/bluetooth/controller/ll_sw/ull_sync.c
@@ -1080,8 +1080,8 @@ void ull_sync_chm_update(uint8_t sync_handle, uint8_t *acad, uint8_t acad_len)
 int ull_sync_slot_update(struct ll_sync_set *sync, uint32_t slot_plus_us,
 			 uint32_t slot_minus_us)
 {
+	uint32_t volatile ret_cb;
 	uint32_t ret;
-	uint32_t ret_cb;
 
 	ret_cb = TICKER_STATUS_BUSY;
 	ret = ticker_update(TICKER_INSTANCE_ID_CTLR,
@@ -1089,8 +1089,8 @@ int ull_sync_slot_update(struct ll_sync_set *sync, uint32_t slot_plus_us,
 			    (TICKER_ID_SCAN_SYNC_BASE +
 			    ull_sync_handle_get(sync)),
 			    0, 0,
-			    slot_plus_us,
-			    slot_minus_us,
+			    HAL_TICKER_US_TO_TICKS(slot_plus_us),
+			    HAL_TICKER_US_TO_TICKS(slot_minus_us),
 			    0, 0,
 			    ticker_update_op_status_give,
 			    (void *)&ret_cb);

